### ISSUE TITLE
No way to query existence of package on remote using profiles

### ISSUE THREAD

**Issue by tmwa** [conan-io/conan#5431]
I would like to be able to check whether a binary package for a particular set of settings and options exists on a remote.
As of Conan 1.16.1 there is no way to do this. There are two approaches that almost work:

```sh
conan install -r remote Package/1.0.0@user/channel -pr profileA -pr profile B
```
* Settings and options can be specified through profiles (good)
* Exit code is nonzero if binary package doesn't exist on remote (good)
* Binary package has to actually be downloaded; there is no flag such as `--dry-run` which would simply report if it exists

```sh
conan search -r remote Package/1.0.0@user/channel --query "os=Linux AND compiler=gcc AND compiler.version=5 AND option1=True AND option2=CustomValue
```
* Does not support settings and options from profiles; all this information has to be replicated in the query syntax, which is cumbersome
* Exit code is always 0, even if no packages matching the query are found

- [x] I've read the [CONTRIBUTING guide](https://github.com/conan-io/conan/blob/develop/.github/CONTRIBUTING.md).
- [x] I've specified the Conan version, operating system version and any tool that can be relevant.
- [x] I've explained the steps to reproduce the error or the motivation/use case of the question/suggestion.

**Comment by memsharded** [comment]
Also, a close alternative to ``install`` would be ``info`` which in theory it is a ``install`` that does not actually install binaries, so it would be equivalent to ``dry-run`` functionality. 

So:
```bash
conan info boost/1.69.0@conan/stable -r=conan-center -s compiler=qcc -s compiler.libcxx=gpp -s compiler.version=4.4
boost/1.69.0@conan/stable
    ID: 5c1390cbdb5a2bbbd9562d0dbe8333a3234205d6
    ...
    Binary: Missing
```
Will produce an output telling that the binary is missing, but yes, it will return 0, not an error.

Same with ``--json`` would produce a machine readable json with ``{"build_id": null, "binary": "Missing"`` field.

I think this is a reasonable use case, and we need to think of some UI if possible to simplify access to this information.

**Comment by memsharded** [comment]
First step should be investigating and discussing what would be the best UI for this use case.

**Comment by AbrilRBS** [comment]
Closing as solved in Conan v2. The `conan list` command is able to list packages for the passed profiles, `conan graph info`, which is able to report the whole status of the dependency graph for one or multiple references for the passed profiles, and `conan graph explain`, which is able to print information regarding the reasons why a certain binary is missing :)

**Comment by AbrilRBS** [comment]
Reopening to track progress for adding profile passing for `conan list`, which turns out I was misremembering with internal development progress, and does not have profile inputs (yet!). Thanks @memsharded for heads-up :)

**Comment by memsharded** [comment]
Closed by https://github.com/conan-io/conan/pull/15697, the ``--filter-profile, --filter-settings and --filter-options`` will be available in Conan 2.2 to ``conan list``
